workloads/pvcscale.yml

```diff
@@ -0,0 +1,141 @@
+
+    - name: Copy workload files
+      copy:
+        src: "{{item.src}}"
+        dest: "{{item.dest}}"
+      with_items:
+        - src: scale-ci-tooling-ns.yml
+          dest: "{{ansible_user_dir}}/scale-ci-tooling/scale-ci-tooling-ns.yml"
+
+    - name: Slurp kubeconfig file
+      slurp:
+        src: "{{kubeconfig_file}}"
+      register: kubeconfig_file_slurp
+
+    - name: Slurp ssh private key file
+      slurp:
+        src: "{{pbench_ssh_private_key_file}}"
+      register: pbench_ssh_private_key_file_slurp
+
+    - name: Slurp ssh public key file
+      slurp:
+        src: "{{pbench_ssh_public_key_file}}"
+      register: pbench_ssh_public_key_file_slurp
+
+    - name: Template workload templates
+      template:
+        src: "{{item.src}}"
+        dest: "{{item.dest}}"
+      with_items:
+        - src: pbench-cm.yml.j2
+          dest: "{{ansible_user_dir}}/scale-ci-tooling/pbench-cm.yml"
+        - src: pbench-ssh-secret.yml.j2
+          dest: "{{ansible_user_dir}}/scale-ci-tooling/pbench-ssh-secret.yml"
+        - src: kubeconfig-secret.yml.j2
+          dest: "{{ansible_user_dir}}/scale-ci-tooling/kubeconfig-secret.yml"
+        - src: workload-job.yml.j2
+          dest: "{{ansible_user_dir}}/scale-ci-tooling/workload-job.yml"
+        - src: workload-pvcscale-script-cm.yml.j2
+          dest: "{{ansible_user_dir}}/scale-ci-tooling/workload-pvcscale-script-cm.yml"
+
+    - name: Check if scale-ci-tooling namespace exists
+      shell: |
+        oc get projects | grep "scale-ci-tooling"
+      ignore_errors: true
+      changed_when: false
+      register: scale_ci_tooling_ns_exists
+    - name: print status of scale_ci_tooling_ns_exists
+      debug:
+        var: scale_ci_tooling_ns_exists
+
+    - name: Ensure any stale scale-ci-pvcscale job is deleted
+      shell: |
+        oc delete job scale-ci-pvcscale -n scale-ci-tooling
+      register: scale_ci_tooling_project
+      failed_when: scale_ci_tooling_project.rc == 0
+      until: scale_ci_tooling_project.rc == 1
+      retries: 60
+      delay: 1
+      when: scale_ci_tooling_ns_exists.rc == 0
+
+    - name: Block for non-existing tooling namespace
+      block:
+        - name: Create tooling namespace
+          shell: |
+            oc create -f {{ansible_user_dir}}/scale-ci-tooling/scale-ci-tooling-ns.yml
+
+        - name: Create tooling service account
+          shell: |
+            oc create serviceaccount useroot -n scale-ci-tooling
+            oc adm policy add-scc-to-user privileged -z useroot -n scale-ci-tooling
+          when: enable_pbench_agents|bool
+      when: scale_ci_tooling_ns_exists.rc != 0
+
+    - name: PVC test specific user configuration 
+      block:
+        - name: Add htpasswd identity provider for user when running pvc scale test
+          shell: |
+            htpasswd -c -B -b /tmp/htpasswd useroot useroot
+            oc create secret generic htpass-secret --from-file=/tmp/htpasswd -n openshift-config
```

Comment:
While running/rerunning this I ran into this error related to the secret created here:
```
FAILED! => {"changed": true, "cmd": "htpasswd -c -B -b /tmp/htpasswd useroot useroot\noc create secret generic htpass-secret --from-file=/tmp/htpasswd -n openshift-config\n", "delta": "0:00:00.379115", "end": "2019-06-14 22:19:38.052228", "msg": "non-zero return code", "rc": 1, "start": "2019-06-14 22:19:37.673113", "stderr": "Adding password for user useroot\nError from server (AlreadyExists): secrets \"htpass-secret\" already exists", "stderr_lines": ["Adding password for user useroot", "Error from server (AlreadyExists): secrets \"htpass-secret\" already exists"], "stdout": "", "stdout_lines": []}
```

We should make this a bit more robust incase the secret already exists.

Reply:
nothing created it,  so it should not be there, if it there it will be used,

Reply:
However the issue is I wanted to run more than once, the task should permit "rerunning".

Reply:
it should delete secret at end of test https://github.com/openshift-scale/workloads/blob/b6e7b19d0c242d8e0e89a8e033499460a88c40f4/workloads/pvcscale.yml#L138 so next run will create it again, but I will experiment a bit more and see what is best

Reply:
added ` oc delete secret -n openshift-config htpass-secret` prior create step,this will ensure secret is created always (in case test did nto reach till end where it is anyway deleted ). I am investigating better way to handle this,will created new PR for this.